synapse/config/appservice.py

```diff
@@ -36,11 +36,10 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
         if not isinstance(self.app_service_config_files, list) or not all(
             type(x) is str for x in self.app_service_config_files
         ):
-            # type-ignore: this function gets arbitrary json value; we do use this path.
             raise ConfigError(
```

Comment:
Can we update `ConfigError` to take `StrCollection` to avoid this in the future?

Reply:
I've ended up getting distracted and doing it anyway. I'll file it as a separate PR.

Reply:
See #15615.